emacs_lua_complete.lua

```diff
@@ -0,0 +1,81 @@
+--    __emacs_lua_complete:
+-- This function finds and prints a limited number of matches on the
+-- lua globals tree, _G. See `lua-complete-string`. N.B.: Does not
+-- complete numeric keys (e.g. for arrays) or keys which begin with an
+-- underscore.
+-- Arguments:
+--   PARTS: a table array of dot-separated parts of the completion item
+--     (e.g. table.con would become {"table","con"}).  All but the last
+--     element must be valid subkeys in the _G heirarchy.
+--   LIBS: an array of tables of the form
+--       {var: varname, lib: libfile}
+--     for e.g. local varname=require("libfile"). This library will be
+--     require'd and searched alongside global variables.
+--   LOCALS: an array of strings parsed from local aloc,bloc = ... to
+--     complete as well
+--
+-- Example:
+-- __emacs_lua_complete({'table','ins'},{{var="xyz",lib="libfile"}},
+--                      {"frank","beans"},5000)
+--
+-- Written 2019/05 J.D. Smith
+function __emacs_lua_complete(parts,libs,locals,limit)
+   -- Setup a FIFO queue for breadth-first tree traversal
+   local queue={head=0,tail=0}
```

Comment:
Coming from Python I feel like this line is missing whitespaces around equals signs (or binary operators). What are the common style guidelines in Lua world? I saw some packages defining whitespace requirements, and some just deferring to "whatever is the default in PIL".

Reply:
lua is very lenient about whitespace in either direction, but I can certainly add it in for visual clarity. Added.  BTW, I will alter history in this branch after the fact to avoid the clutter of small commits like these, so please don't rely on specific commit ID's!

Reply:
Yeah, let's please add whitespace around binary operators and newlines where necessary, now that there's no necessity to keep the size to the minimum.

And things like
```
    local globals,cnt,limit={},0,limit or 2000 
```
could probably use a touch to clarify (`limit` assignment in particular catches me off guard every time I return to this piece of code).